File: test-prng.js
#!/usr/bin/env node

// Test to understand the PRNG mismatch between Rust and JavaScript

const {PRNG} = require('./../pokemon-showdown-ts/dist/sim/prng');

console.log('=== Testing JavaScript PRNG ===');
console.log('Initial seed: [0, 0, 0, 1]');

const prng = new PRNG([0, 0, 0, 1]);

console.log('\nFirst 10 random() calls:');
for (let i = 0; i < 10; i++) {
	const val = prng.random(100);
	console.log(`${i}: ${val}`);
}

console.log('\n=== Testing with fresh PRNG ===');
const prng2 = new PRNG([0, 0, 0, 1]);
console.log('First random_range(50, 101):', prng2.random(50, 101));
console.log('Second random_range(50, 101):', prng2.random(50, 101));

File: examples/test_prng.rs
/// Simple test to trace PRNG behavior

use pokemon_showdown::{PRNG, PRNGSeed};

fn main() {
    println!("=== Testing Rust PRNG ===");
    println!("Initial seed: [0, 0, 0, 1]");

    let mut prng = PRNG::new(Some(PRNGSeed::Gen5([0, 0, 0, 1])));

    println!("\nFirst 10 random_range(0, 100) calls:");
    for i in 0..10 {
        let val = prng.random_range(0, 100);
        println!("{}: {}", i, val);
    }

    println!("\n=== Testing with fresh PRNG ===");
    let mut prng2 = PRNG::new(Some(PRNGSeed::Gen5([0, 0, 0, 1])));
    println!("First random_range(50, 101): {}", prng2.random_range(50, 101));
    println!("Second random_range(50, 101): {}", prng2.random_range(50, 101));
}
